spec/models/user_spec.rb

```diff
@@ -0,0 +1,13 @@
+require "rails_helper"
+require "spec_helper"
+
+RSpec.describe User, type: :model do
+  describe "validations of presence and uniqueness" do
+    it { is_expected.to validate_presence_of(:username) }
+    it { is_expected.to validate_presence_of(:email) }
```

Comment:
Add test for valid invalid email

Reply:
added

Reply:
in that block all test cases will use that subject. position does not matter inside the block